protocol/v2/types/batch_verifier.go

```diff
@@ -0,0 +1,475 @@
+package types
```

Comment:
Generally its hard for me to understand the gain we get from this batchverifier without comparison to previous state. also a lot of locks and stuff here, I think we need to revisit this code and optimize.

Reply:
@moshe-blox FYI: I'm going to delete the batch verifier because it doesn't pass the review. If we need it in the future we'll be able to copy it from this PR

Reply:
@y0sher deleted

Reply:
oh I didn't know its out of use for now. so ignore all my comments on the batch verifier.